diff/diff.go

```diff
@@ -19,6 +21,91 @@ import (
 	"github.com/kong/go-kong/kong"
 )
 
+type EntityState struct {
+	Name string `json:"name"`
+	Kind string `json:"kind"`
+	Body any    `json:"body"`
+}
+
+func diffObjects(obj1, obj2 interface{}) interface{} {
```

Comment:
Can you please add a docstring to this function?

Also, can you please add unit tests?

Reply:
done.